plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/NameEmbeddings.kt

```diff
@@ -28,7 +28,7 @@ data class LocalName(val name: Name) : MangledName {
 
 data class ClassName(val packageName: FqName, val className: Name) : MangledName {
     override val mangled: String
-        get() = "pkg_${packageName.asString()}\$class_${className.asString()}"
```

Comment:
The underscore was actually intentional since it still shouldn't collide and makes the whole thing a bit more readable than a bunch of _, though I agree that if we use $ for the package separator that makes it harder.  Either is fine by me.

Reply:
I like underscores more as well. Problem is just that we need to replace the dots in the package name with something because Viper can't have dots in names. And if we replace them with underscores we could theoretically have name collisions so that's why I went for $.

Reply:
Yeah, I agree for consistency it's probably best.  I wonder whether there's any other symbols that would work...  I can't find any other symbols permitted in Viper identifiers that could be useful, though.  Pity it doesn't support Unicode.

(Also pity that even though it's built to be a VIR it doesn't support some kind of richer notion of names...)